File: kotlin-node/karakum/src/jsMain/resources/plugins/convertURL.js
import ts from "typescript";

export default function (node, context, render) {
    const sourceFileName = node.getSourceFile()?.fileName ?? "generated.d.ts"

    if (
        sourceFileName.endsWith("url.d.ts")
        && ts.isClassDeclaration(node)
        && node.name?.text === "URL"
    ) {
        // remove from conversion to avoid file creation for anonymous declarations
        return ""
    }


    return null
}

File: kotlin-node/karakum/src/jsMain/kotlin/node/karakum/plugins/convertURL.kt
package node.karakum.plugins

import node.karakum.util.nullable
import io.github.sgrishchenko.karakum.extension.createPlugin
import io.github.sgrishchenko.karakum.util.getSourceFileOrNull
import typescript.isClassDeclaration

val convertURL = createPlugin { node, context, render ->
    nullable {
        val sourceFileName = ensureNotNull(node.getSourceFileOrNull()).fileName
        ensure(sourceFileName.endsWith("url.d.ts"))

        ensure(isClassDeclaration(node))
        ensure(node.name?.text == "URL")

        // remove from conversion to avoid file creation for anonymous declarations
        ""
    }
}
